### PR TITLE
[chore] Create configuration to enable development in this repository using Project IDX

### PR DESCRIPTION
**Pull request by michaelsafyan**
**Description:**

Specifies the necessary `.idx/...` config so that this repository (or forks of this repository) can easily be edited in IDX.

**Purpose**

Increase development velocity.

**Background:**

[Project IDX](https://idx.dev/) is a Cloud-based IDE from Google. It provides for a hermetic, reproducible development environment using [NixOS](https://nixos.org/) configuration to define environment dependencies. Development with Project IDX requires a `.idx/dev.nix` configuration to define the requires libraries, binaries, packages, services, etc.

For more info, see:

 - https://developers.google.com/idx

### CODE REVIEW

**Review by mx-psi — CHANGES_REQUESTED** [review]
Hey, thanks for your contribution! While I understand how this may be useful for you and other folks using IDX, it would be hard for us to maintain this, since (AFAIK) none of the maintainers use it. We could improve this with some sort of CI check, but still we need to justify that this is used.

I would suggest something like this:
- File a PR that adds `.idx/` to `.gitignore`, so that you and other users can add their own configuration comfortably
- File an issue to gauge interest through upvotes to add this and close this PR for now

**Comment by michaelsafyan** [comment]
Putting it in `.gitignore` and reverting is not ideal for several reasons:

 1. Creating a correct `.idx/dev.nix` takes time.
 2. Different versions of `.idx/dev.nix` defeats the main benefit of a reproducible/repeatable environment.
 3. A copy is needed in each branch, adding work effort/friction to usage.
 4. That solution would also preclude ["Open in IDX"](https://developers.google.com/idx/guides/open-in-idx).

If no one ends up using IDX for development with this repository, then it is also of no consequence if it breaks; breakage in the configuration here would only affect those developing in this repository using IDX; it would not negatively impact those making use of the components provided in this repository. Worst-case scenario: it breaks, and then the next person who wants to use it with IDX would notice that it doesn't work and would need to fix it.

**Comment by mx-psi** [comment]
I understand that `.gitignore` is not ideal, but this is something that requires no maintenance from our side and solves at least one problem (that of needing to remove this folder before commiting). Adding this to `.gitignore` does not mean we can't eventually add `.idx` to our codebase if enough people want it and we know how to maintain it.

I still don't think it's a good idea to commit something that we don't know how to maintain. We (maintainers, approvers) won't be able to address issues raised about it, which is one of the responsibilities assumed of us.

If, after opening an issue, this gains enough traction, then we can consider it, but right now we don't have a good idea of how many people this would be useful to.

cc @open-telemetry/collector-approvers to chime in in case you disagree with this assessment

File: .idx/dev.nix
Comment (atoulme) [inline .idx/dev.nix:21]: We don't allow CGO for the collector right now.

**Comment by michaelsafyan** [comment]
At head, "make test" fails without this flag, which suggests that one of
the components in the repository is using this setting.

**Comment by mx-psi** [comment]
> At head, "make test" fails without this flag, which suggests that one of the components in the repository is using this setting.

This is likely because the race detector we use on tests needs cgo. But on actual builds we forbid cgo entirely

**Review by TylerHelmuth — CHANGES_REQUESTED** [review]
I agree that we should not merge this. The repository should not start taking on IDE-specific files as we won't be able to validate them or how they'll interact with other IDEs.

**Comment by michaelsafyan** [comment]
Thanks for the feedback.

I have sent a separate pull request to add `.idx/` to `.gitignore`:

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34457

**Comment by michaelsafyan** [comment]
Given that there are now two expressions of concern regarding submitting such a config, would it still be useful/productive to create "an issue to gauge interest through upvotes" as initially suggested? Or is that a lost cause?

**Comment by mx-psi** [comment]
> Given that there are now two expressions of concern regarding submitting such a config, would it still be useful/productive to create "an issue to gauge interest through upvotes" as initially suggested? Or is that a lost cause?

We would need a significant number of people to express a need/desire for this, as well as a plan on how we can maintain this before adding it. I personally would not outright discard the possibility if IDX becomes wildly popular in the future, but we can wait for @TylerHelmuth to give their opinion as well before filing the issue

**Comment by TylerHelmuth** [comment]
Issue for discussion is fine.

**Pull request closed without merging**
